crates/query/src/scan/array_predicate.rs

```diff
@@ -409,4 +409,49 @@ impl ArrayPredicate for InList {
         let mask = sqd_polars::arrow::polars_boolean_to_arrow_boolean(&polars_mask);
         Ok(mask)
     }
-}
\ No newline at end of file
+}
+
+
+pub struct BloomFilter {
+    list: sqd_polars::prelude::Series
+}
+
+
+impl BloomFilter {
+    pub fn new<T: IntoArrow>(values: Vec<T>) -> Self {
+        let arr = T::make_array(values);
+        let list = sqd_polars::arrow::array_series("value_list", &arr).unwrap();
+        Self {
+            list
+        }
+    }
+
+    pub fn bloom_contains(&self, opt: Option<sqd_polars::prelude::Series>) -> anyhow::Result<bool> {
+        if let Some(val) = opt {
+            let bit_array: Vec<_> = val.bool()?.iter().map(|opt| opt.unwrap()).collect();
+            let bloom = sqd_bloom_filter::BloomFilter::from_bit_array(bit_array, 7);
```

Comment:
@eldargab would it make sense to import num hashes (7) as `sqd_data::solana::tables::instruction::NUM_HASHES`. it would require adding sqd_data as a dep for this crate

Reply:
I don't see any problems with depending on `sqd_data`

Reply:
i decided to use hardcoded 64 and 7. i believe BloomFilter predicate should be more generic and have no dependencies from smth specific like solana instructions... but let's keep it until we need another bloom filter field?